src/app/trip-page/trip-photos-list/trip-photos-list.component.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Input,
+  OnInit,
+  QueryList,
+  ViewChildren,
+} from '@angular/core';
+import { BehaviorSubject, Observable, of, Subject, take } from 'rxjs';
+import { TripPhoto, TripPhotoService } from 'src/app/shared';
+
+@Component({
+  selector: 'app-trip-photos-list',
+  templateUrl: './trip-photos-list.component.html',
+  styleUrls: ['./trip-photos-list.component.css'],
+})
+export class TripPhotosListComponent {
+  initTripPhoto: TripPhoto = {
+    id: '',
+    tripId: 0,
+    fileName: '',
+    fileType: '',
+    path: '',
+    addedByUser: '',
+    alt: '',
+  }
+  @Input() tripPhotos$: Observable<TripPhoto[]> = of([]);
+  @ViewChildren('photoselect') private photos!: QueryList<ElementRef>;
+  progressMessage$: BehaviorSubject<string> = new BehaviorSubject('');
+  selectedCount$: BehaviorSubject<number> = new BehaviorSubject(0);
```

Comment:
BehaviorSubjects should be private. If you want to expose the observable part of that to the view you should create a public Observable from the BehaviorSubject with `asObservable`

Reply:
I was able to get it to work with a private behavior subject in the trip photos service and using a method to update it and an observable to access the behavior subject value from components. However, I realized for my use case I was wrong in originally using behavior subjects for child to parent  & parent to child updates so I removed them and just used observables. 

sidenote: I was able to remove a bit of change detection, but not all: I tried different things to try to get it to update the `tripPhotos$` observable between components upon photo file upload/deletion but wasn't able to get it to update w/o calling `detectChanges()`.